lib/printer.rb

```diff
@@ -1,17 +1,32 @@
 class Printer
-  attr_reader :stdout, :grid
+  attr_reader :stdout
 
-  def initialize(stdout, grid)
+  def initialize(stdout)
     @stdout = stdout
-    @grid = grid
   end
 
   def print_welcome_message
-    stdout.puts(grid.template)
     stdout.puts("Enter your move >")
   end
 
+  def print_updated_grid(coordinates)
+    stdout.puts(print_current_grid(coordinates))
```

Comment:
There is an interesting naming smell here. In other print methods you pass puts a string. But in this one you pass it a method with the name `print` in it, even though it doesn't actually print.

Unlike all the other methods the `print_current_grid` is not a print method. It is a template/data method.